RockPaperScissors/RockPaperScissors.cs

```diff
@@ -1,18 +1,57 @@
 ﻿using System;
 
+
 namespace RockPaperScissors
 {
     class Program
     {
         public static void Main()
         {
-            Console.WriteLine("Enter your hand:");
-            string hand1 = Console.ReadLine().ToLower();
-            string hand2 = ComputerHandGenerator();
-            Console.WriteLine(CompareHands(hand1, hand2));
+            string hand1 = "";
+            bool endGame = false;
+            //Read user's input
+
+
+            //Try-Catch-Finally for error handling
+            while (endGame == false)
+            {
+                Console.WriteLine("Enter your hand:");
+                try
+                {
+                    hand1 = Console.ReadLine().ToLower();
+                }
+                catch
+                {
+                    Console.WriteLine("Error. Please enter a valid hand ('rock','paper','scissors')!");
```

Comment:
You would catch the exception here if one was thrown; however, all you do is write to the screen. So if you end the wrong thing you still fall threw to the finally and print `Displaying results....`. See if you can rework this to allow the user to reenter their play if it is invalid.